### ISSUE TITLE
[Bug] CUrl bug, urls with two slashes at domain level are stripped out, causing download error

### ISSUE THREAD

**Issue by lavish440** [xbmc/xbmc#25666]
### Describe the problem

I am trying to play a stream which has two `//` in the URL. It used to work fine earlier in Kodi. 
But now ISA treats `//` as `/` causing the stream server to give 404.
It works perfectly in other players like mpv and vlc.

### Steps to reproduce

1. Try playing a stream which has two `//` in the URL

### Debug log

[Log](https://paste.kodi.tv/uhonacicam.kodi)
Some URLs have been redacted/replaced.

### Additional info

One line in my addon code before providing the URL to ISA, I am using Script.notify to display it to me. It shows two slashes there.
The log also says it has two slashes but when I use a proxy to intercept the traffic, ISA URL only has a single slash but the ACL param in the URL contains two // which is the correct behavior for the ACL param.
![HTTPS Proxy](https://github.com/user-attachments/assets/6b688bf1-c20f-46ec-9880-9aa118cfdac4)

The first request in the screenshot is by my addon to request the available playback URLs.
The second request I am making with `urlquick` for some debugging and the request is successfully made.
The third request is by ISA which fails.
NOTE: ISA and urlquick are provided the same URL in the addon.

### Operating system(s)

Android, Linux

### Operating system version(s)

Android 14, Arch Linux

### InputStream Adaptive version(s)

22.1.2 (Compiled from Piers branch)

### Kodi version(s)

22.0-ALPHA1 (21.90.700) Git:20240820-178dc39-dirty (Compiled from Master branch)

**Comment by lavish440** [comment]
> So it is slashes, not backslashes.

I have corrected it in the title now.

**Comment by matthuisman** [comment]
A workaround may be to url encode the 2nd / (%2F)

**Comment by lavish440** [comment]
> A workaround may be to url encode the 2nd / (%2F)

Url-encoding the second '/' does not work

**Comment by CastagnaIT** [comment]
@lavish440 i made a fix
when you fnd time please give a feedback

here test builds, notice that are Kodi 22 nightly
armv7 https://jenkins.kodi.tv/job/android-arm-docker/8117/artifact/kodi-20241008-0ca033f0-PR25808-armeabi-v7a.apk
arm64 https://jenkins.kodi.tv/job/android-arm64-docker/7909/artifact/kodi-20241008-0ca033f0-PR25808-arm64-v8a.apk

**Comment by lavish440** [comment]
> @lavish440 i made a fix

But I can only test on Linux currently and when I try to compile Kodi, it fails on Arch Linux.

**Comment by CastagnaIT** [comment]
im not so linux expert, but at least on ubuntu i have no problem to build it,
try look steps at https://github.com/xbmc/xbmc/blob/master/docs/README.Linux.md
its possible missing some required package

**Comment by lavish440** [comment]
> im not so linux expert, but at least on ubuntu i have no problem to build it,
> try look steps at https://github.com/xbmc/xbmc/blob/master/docs/README.Linux.md
> its possible missing some required package

Let me try again. I am arch so the packages are different.

**Comment by lavish440** [comment]
@CastagnaIT Tried everything I could to build on my laptop. Aur, git etc..
Still no avail. 
I spent at least 3-4 hours building it but still it fails at one step or another.

**Comment by basilgello** [comment]
@graysky2

**Comment by CastagnaIT** [comment]
can you copy paste the errors?

**Comment by lavish440** [comment]
Earlier the errors were related to CD and DVD something. 
So it compiled without CD and DVD support. 
Now ffmpeg is the issue

**Comment by lavish440** [comment]
> you have mentioned also Android above, you cannot test on android?

Ohh sorry. 
I forgot about that. Let me try again but I can't intercept the communication like on my Laptop because I can't install the https cert in Kodi due to Android 14 restrictions.

**Comment by CastagnaIT** [comment]
build artifacts expire after some days
new links:
https://jenkins.kodi.tv/job/android-arm-docker/8226/artifact/kodi-20241018-88acea54-PR25808-armeabi-v7a.apk
https://jenkins.kodi.tv/job/android-arm64-docker/8007/artifact/kodi-20241018-88acea54-PR25808-arm64-v8a.apk

**Comment by lavish440** [comment]
Seems like `script.module.distutils` is not available in this Kodi version.

**Comment by lavish440** [comment]
Looks like my provider has removed those two concurrent slashes from the stream URL. I will try to test with a test server.

**Comment by lavish440** [comment]
Tried a test stream served by NodeJS Express and I can confirm that the issue has been fixed in this test build. 

Also, I have noticed another related issue. 
For example, if a stream is running on a non-standard port (other than 80 kr 443) let's say 8080, if ISA makes a get request to the manifest and receives a redirect and a path in the location header, it will omit the port when try to access the new URL. This doesn't happen when using InputStream FFmpeg Direct. 

Example -
Orginal URL -> https://example.com:8080/1.dash
Server replies with -> `Location: /newPath/1.dash`
ISA tries to get https://example.com/newPath/1.dash omitting the port.

**Comment by CastagnaIT** [comment]
redirects are managed by kodi core not ISA
ISA do a single request and receive the response have no others "options" from the kodi binary interface,
then there could be another bug in to curl of kodi core

if you think it needs to be fixed open a separate Issue for this use case and if possible a way to reproduce it locally

anyway using a manifest url with ports specified its not so normal thing,
iirc ffmpeg should have its internal library for http, should not use curl of kodi core
